src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml

```diff
@@ -449,6 +520,54 @@ components:
           type: string
       required:
         - status
+    AuditedResource:
```

Comment:
I've created a new type, which is essentially a base class / super class.
We've copy and pasted these 8 fields into pretty much every response type, so in an attempt to reduce the size of the swagger spec file I've introduced this, then will reference it where we use those fields.
At the moment I've only referenced it in the new response types I've introduced here, plus one other place as an example, but we can create future PRs to use this type as a base type over time 👍

Reply:
I'm not sure how this translates into the generated code if this is just a swagger optimisation and it comes out flattened then I think it's fine. Otherwise I would suggest we should favour composition over inheritance and we could have this as it's own type with an `auditInfo` or similar field on each object.

Reply:
In respect of the generated code, I guess it's loosely inheritance, but without the formality of extending a class. Using your term, the generated types are "flattened" - eg:
```
package uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
    import jakarta.validation.constraints.*
    import jakarta.validation.Valid
    import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.resource.validator.*
    import io.swagger.v3.oas.annotations.media.Schema

        /**
 * Represents a Conversation with a prisoner within the context of their Learning and Work Progress.
 * @param reference A unique reference reference for this ConversationResponse.
 * @param prisonNumber The ID of the prisoner
 * @param note A note for this conversation
 * @param createdBy The DPS username of the person who created this resource.
 * @param createdByDisplayName The display name of the person who created this resource.
 * @param createdAt An ISO-8601 timestamp representing when this resource was created.
 * @param createdAtPrison The identifier of the prison that the prisoner was resident at when this resource was created.
 * @param updatedBy The DPS username of the person who last updated this resource.
 * @param updatedByDisplayName The display name of the person who last updated this resource.
 * @param updatedAt An ISO-8601 timestamp representing when this resource was last updated. This will be the same as the created date if it has not yet been updated.
 * @param updatedAtPrison The identifier of the prison that the prisoner was resident at when this resource was updated.
 */
data class ConversationResponse(

    @Schema(example = "814ade0a-a3b2-46a3-862f-79211ba13f7b", required = true, description = "A unique reference reference for this ConversationResponse.")
    @get:JsonProperty("reference", required = true) val reference: java.util.UUID,

    @Schema(example = "A1234BC", required = true, description = "The ID of the prisoner")
    @get:JsonProperty("prisonNumber", required = true) val prisonNumber: kotlin.String,

    @Schema(example = "Pay close attention to Peter's behaviour.", required = true, description = "A note for this conversation")
    @get:JsonProperty("note", required = true) val note: kotlin.String,

    @Schema(example = "asmith_gen", required = true, description = "The DPS username of the person who created this resource.")
    @get:JsonProperty("createdBy", required = true) val createdBy: kotlin.String,

    @Schema(example = "Alex Smith", required = true, description = "The display name of the person who created this resource.")
    @get:JsonProperty("createdByDisplayName", required = true) val createdByDisplayName: kotlin.String,

    @Schema(example = "2023-06-19T09:39:44Z", required = true, description = "An ISO-8601 timestamp representing when this resource was created.")
    @get:JsonProperty("createdAt", required = true) val createdAt: java.time.OffsetDateTime,

    @Schema(example = "BXI", required = true, description = "The identifier of the prison that the prisoner was resident at when this resource was created.")
    @get:JsonProperty("createdAtPrison", required = true) val createdAtPrison: kotlin.String,

    @Schema(example = "asmith_gen", required = true, description = "The DPS username of the person who last updated this resource.")
    @get:JsonProperty("updatedBy", required = true) val updatedBy: kotlin.String,

    @Schema(example = "Alex Smith", required = true, description = "The display name of the person who last updated this resource.")
    @get:JsonProperty("updatedByDisplayName", required = true) val updatedByDisplayName: kotlin.String,

    @Schema(example = "2023-06-19T09:39:44Z", required = true, description = "An ISO-8601 timestamp representing when this resource was last updated. This will be the same as the created date if it has not yet been updated.")
    @get:JsonProperty("updatedAt", required = true) val updatedAt: java.time.OffsetDateTime,

    @Schema(example = "BXI", required = true, description = "The identifier of the prison that the prisoner was resident at when this resource was updated.")
    @get:JsonProperty("updatedAtPrison", required = true) val updatedAtPrison: kotlin.String
) {

}
```

Whilst this is working for simple types like `ConversationResponse` it's not working in the array structure (which is where I really wanted it! 🤣 )
I need to play with a few options .... (and the discussion about API naming, and whether this API is just "pre-prison" or all quals will play into this)